sdk/eventhub/Azure.Messaging.EventHubs.Processor/samples/Sample08_MockingClientTypes.md

```diff
@@ -0,0 +1,68 @@
+# Mocking Client Types
+
+Event Hubs is built to support unit testing with mocks, as described in the [Azure SDK Design Guidelines](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-mocking). This is an important feature of the library that allows developers to write tests that are completely focused on their own application logic, though they depend on the Event Hubs types.
+
+For a more detailed description of mocking the set of Event Hubs client types see TODO INSERT LINK TO OTHER SAMPLE ONCE MERGED.
+
+## Application-defined handlers
+
+Interacting with an `EventHubProcessorClient` in an application is done through various handler methods. The Event Hubs library guarantees that each of these handlers will be called at the appropriate times. Therefore, when validating correctness of an application dependent on an `EventHubProcessorClient`, each handler can just be called and tested directly.
```

Comment:
```suggestion
Interacting with an `EventHubProcessorClient` in an application is done through various handler methods. The Event Hubs library guarantees that each of these handlers will be called at the appropriate times while the processor is running. Therefore, it is recommended that handers be tested by calling them directly, rather than attempting to simulate having the processor producer invoke them.
```